dotnet/samples/KernelSyntaxExamples/Example22_OpenAIPlugin_AzureKeyVault.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Mime;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Functions.OpenAPI.Authentication;
+using Microsoft.SemanticKernel.Functions.OpenAPI.Model;
+using Microsoft.SemanticKernel.Functions.OpenAPI.OpenAI;
+using Microsoft.SemanticKernel.Functions.OpenAPI.Plugins;
+using Microsoft.SemanticKernel.Orchestration;
+using RepoUtils;
+using Resources;
+
+// ReSharper disable once InconsistentNaming
+public static class Example22_OpenAIPlugin_AzureKeyVault
+{
+    private const string SecretName = "Foo";
+    private const string SecretValue = "Bar";
+
+    /// <summary>
+    /// This example demonstrates how to connect an Azure Key Vault plugin to the Semantic Kernel.
+    /// To use this example, there are a few requirements:
+    ///   1. Register a client application with the Microsoft identity platform.
+    ///   https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app
+    ///
+    ///   2. Create an Azure Key Vault
+    ///   https://learn.microsoft.com/en-us/azure/key-vault/general/quick-create-portal
+    ///
+    ///   3. Add a permission for Azure Key Vault to your client application
+    ///   https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-configure-app-access-web-apis
+    ///
+    ///   4. Set your Key Vault endpoint, client ID, and client secret as user secrets using:
+    ///   dotnet user-secrets set "KeyVault:Endpoint" "your_endpoint"
+    ///   dotnet user-secrets set "KeyVault:ClientId" "your_client_id"
+    ///   dotnet user-secrets set "KeyVault:ClientSecret" "your_secret"
+    ///
+    ///   5. Replace your tenant ID with the "TENANT_ID" placeholder in dotnet/samples/KernelSyntaxExamples/Resources/22-ai-plugin.json
+    /// </summary>
+    public static async Task RunAsync()
+    {
+        var authenticationProvider = new OpenAIAuthenticationProvider(
+            new Dictionary<string, Dictionary<string, string>>()
+            {
+                {
+                    "login.microsoftonline.com",
+                    new Dictionary<string, string>()
+                    {
+                        { "client_id", TestConfiguration.KeyVault.ClientId },
+                        { "client_secret", TestConfiguration.KeyVault.ClientSecret },
+                        { "grant_type", "client_credentials" }
+                    }
+                }
+            }
+        );
+
+        var kernel = new KernelBuilder().WithLoggerFactory(ConsoleLogger.LoggerFactory).Build();
+
+        var openApiSpec = EmbeddedResource.Read("22-openapi.json");
+        var messageStub = new HttpMessageHandlerStub(openApiSpec);
```

Comment:
nit: add using for both stub and client instead of disposing them manually it at the end of the method.